test/e2eTests/pages/page.js

```diff
@@ -11,41 +12,51 @@ const Component = require('./components/component')
 // For example, when the patientSearch component is added to a page
 // tests can call page.search(...)
 class Page {
-	constructor(options) {
-		// Make sure the components array is inialized
-		options.components = options.components || [];
-
-		this.options = options;
-
-		// Every page should have the header component
-		if(!this.options.components.includes('header')) {
-			this.options.components.push('header');
-		}
-	}
-
-	_init() {
-		this.I = actor();
-
-		// Add each component to this page
-		this.options.components.forEach(name => this._addComponent(name));
-	}
-
-	// Validates that the page is loaded
-	isLoaded() {
-		this.I.waitForElement(this.options.isLoaded.element, 5);
-		this.I.seeInCurrentUrl(this.options.isLoaded.urlPart);
-
-		// If there is an overlay, wait for it
-		this.I.waitForInvisible('#overlay', 10);
-	}
-
-	// Gets an instance of the component
-	// and copies its properties and functions
-	// to this page
-	_addComponent(componentName) {
-		const component = Component.create(componentName);
-		component.addToPage(this);
-	}
+  constructor(options) {
+    // Make sure the components array is inialized
+    options.components = options.components || [];
+
+    this.options = options;
+
+    // Every page should have the header component
+    if (!this.options.components.includes('header')) {
+      this.options.components.push('header');
+    }
+
+    // Add translator
+    this.i18n = new I18n({
+      directory: __dirname + '/../../../poc_config/openmrs/i18n/'
+    });
+  }
+
+  _init() {
+    this.I = actor();
+
+    // Add each component to this page
+    this.options.components.forEach(name => this._addComponent(name));
+  }
+
+  // Validates that the page is loaded
+  isLoaded() {
+    this.I.waitForElement(this.options.isLoaded.element, 5);
+    this.I.seeInCurrentUrl(this.options.isLoaded.urlPart);
+
+    // If there is an overlay, wait for it
+    this.I.waitForInvisible('#overlay', 10);
+  }
+
+  // Gets an instance of the component
+  // and copies its properties and functions
+  // to this page
+  _addComponent(componentName) {
+    const component = Component.create(componentName);
+    component.addToPage(this);
+  }
+
+  // TODO: Set locale automatically
+  translate(key) {
+    return this.i18n.t('locale_pt', key);
```

Comment:
It could be in the _init but we may need it in the sub classes constructor. For instance I locate a button in RegistrationDashboardPage constructor using the translate method.
It's a good point. Perhaps finding another way to locate the elements.

Reply:
Alternatively, the translator could be a singleton that any class can reference by a public static instance.